sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExec.scala

```diff
@@ -48,9 +49,6 @@ case class ShowTablesExec(
   }
 
   private def isTempView(ident: Identifier): Boolean = {
-    catalog match {
-      case s: V2SessionCatalog => s.isTempView(ident)
-      case _ => false
-    }
+    session.sessionState.catalog.isTempView((ident.namespace() :+ ident.name()).toImmutableArraySeq)
```

Comment:
shall we only do it if the catalog name is `spark_catalog`? i.e. `CatalogV2Utils.isSessionCatalog`

Reply:
good point! refactored